extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -262,49 +281,87 @@ public static String extractVideoIdFromMixId(final String playlistId) throws Par
 
         } else if (isYoutubeChannelMixId(playlistId)) { // starts with "RMCM"
             // Channel mix are build with RMCM{channelId}, so videoId can't be determined
-            throw new ParsingException("Video id could not be determined from mix id: " + playlistId);
+            throw new ParsingException("Video id could not be determined from mix id: "
+                    + playlistId);
 
         } else if (isYoutubeMixId(playlistId)) { // normal mix, starts with "RD"
             return playlistId.substring(2);
 
         } else { // not a mix
-            throw new ParsingException("Video id could not be determined from mix id: " + playlistId);
+            throw new ParsingException("Video id could not be determined from mix id: "
+                    + playlistId);
         }
     }
 
     public static JsonObject getInitialData(final String html) throws ParsingException {
         try {
             try {
-                final String initialData = Parser.matchGroup1("window\\[\"ytInitialData\"\\]\\s*=\\s*(\\{.*?\\});", html);
+                final String initialData = Parser.matchGroup1(
+                        "window\\[\"ytInitialData\"\\]\\s*=\\s*(\\{.*?\\});", html);
                 return JsonParser.object().from(initialData);
-            } catch (Parser.RegexException e) {
-                final String initialData = Parser.matchGroup1("var\\s*ytInitialData\\s*=\\s*(\\{.*?\\});", html);
+            } catch (final Parser.RegexException e) {
+                final String initialData = Parser.matchGroup1(
+                        "var\\s*ytInitialData\\s*=\\s*(\\{.*?\\});", html);
                 return JsonParser.object().from(initialData);
             }
-        } catch (JsonParserException | Parser.RegexException e) {
+        } catch (final JsonParserException | Parser.RegexException e) {
             throw new ParsingException("Could not get ytInitialData", e);
         }
     }
 
-    public static boolean isHardcodedClientVersionValid() throws IOException, ExtractionException {
-        final String url = "https://www.youtube.com/results?search_query=test&pbj=1";
+    public static Optional<Boolean> areHardcodedClientVersionAndKeyValid()
```

Comment:
I think you should just return a plain `boolean` from this method, since the returned `Optional` would never be empty anyway (you check for emptiness right at the beginning). This way you can do just `areHardcodedClientVersionAndKeyValid()` instead of `areHardcodedClientVersionAndKeyValid().orElse(false)`

Reply:
Done